staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -100,10 +99,11 @@ type RequestDigest struct {
 // this type and cfgMeal follow the convention that the suffix
 // "Locked" means that the caller must hold the configController lock.
 type configController struct {
-	name             string // varies in tests of fighting controllers
-	clock            clock.PassiveClock
-	queueSetFactory  fq.QueueSetFactory
-	obsPairGenerator metrics.TimedObserverPairGenerator
+	name                  string // varies in tests of fighting controllers
+	clock                 clock.PassiveClock
+	queueSetFactory       fq.QueueSetFactory
+	reqsObsPairGenerator  metrics.TimedObserverPairGenerator
```

Comment:
Watermarking has been there since before I got involved.  I found that the max-in-flight filter was already watermarking the number in flight.  That makes sense, because this is a gauge that can vary much more quickly than we can expect scrapes to happen.

Oddly, the watermarking that the max-in-flight filter has been doing is itself something that surely will not be scraped frequently enough.  The watermarking in that filter is only over the last second, and we certainly can expect the apiserver metrics scraping period to be longer than one second!

We could try to pick a better watermarking period, but that would be a difficult exercise in satisfying everybody when we do not even know who everybody is.  What the watermark histogram does is take observations of watermarks, so that no scraping period misses watermark observations.

Reply:
I don't fully agree, but I don't want to block this PR on it either (sorry for sitting so long on it so far).
It's not introducing anything new - and we should discuss how to improve it separately.

I opened https://github.com/kubernetes/kubernetes/issues/106302 to discuss that further.